pkg/pod/pod.go

```diff
@@ -219,6 +220,17 @@ func MakePod(images pipeline.Images, taskRun *v1beta1.TaskRun, taskSpec v1beta1.
 		return nil, err
 	}
 
+	// Using node affinity on taskRuns sharing PVC workspace, with an Affinity Assistant
+	// is mutual exclusive with other affinity on taskRun pods. If other
+	// affinity is wanted, that should be added on the Affinity Assistant pod unless
+	// assistant is disabled. When Affinity Assistant is disabled, an affinityAssistantName is not set.
+	var affinity *corev1.Affinity
+	if affinityAssistantName := taskRun.Annotations[workspace.AnnotationAffinityAssistantName]; affinityAssistantName != "" {
+		affinity = nodeAffinityUsingAffinityAssistant(affinityAssistantName)
```

Comment:
As @afrittoli mentioned. If `podTemplate.Affinity` has a value here, we can emit a _warning_-Event.